frontend/class-frontend.php

```diff
@@ -1907,20 +1910,22 @@ protected function json_ld_output( $output, $context ) {
 		 *
 		 * @param string $context The context of the output, useful to determine whether to filter or not
 		 */
-		$output = apply_filters( 'wpseo_json_ld_output', $output, $context );
+		$this->output = apply_filters( 'wpseo_json_ld_output', $this->output, $context );
 
-		echo json_encode( $output );
+		if ( ! is_array( $this->output ) || $this->output === array() ) {
```

Comment:
I prefer `! empty()` checks (several more places here too) over comparisons. They are more robust and handle more cases of empty-ness.